applications/accounting/groovyScripts/reports/CostCenters.groovy

```diff
@@ -26,50 +25,49 @@ import org.apache.ofbiz.entity.condition.EntityOperator
+andExprs.add(EntityCondition.makeCondition('transactionDate', EntityOperator.LESS_THAN_EQUAL_TO, thruDate))
 andCond = EntityCondition.makeCondition(andExprs, EntityOperator.AND)
-List postedTransactionTotals = select("glAccountId", "accountName", "accountCode", "debitCreditFlag", "amount").from("AcctgTransEntrySums").where(andCond).orderBy("glAccountId").queryList()
+List postedTransactionTotals = select('glAccountId', 'accountName', 'accountCode', 'debitCreditFlag', 'amount')
+        .from('AcctgTransEntrySums').where(andCond).orderBy('glAccountId').queryList()
 if (postedTransactionTotals) {
-    glAccountCategories = from("GlAccountCategory").where("glAccountCategoryTypeId", "COST_CENTER").orderBy("glAccountCategoryId").queryList()
+    glAccountCategories = from('GlAccountCategory').where('glAccountCategoryTypeId', 'COST_CENTER').orderBy('glAccountCategoryId').queryList()
     context.glAccountCategories = glAccountCategories
     Map postedTransactionTotalsMap = [:]
     postedTransactionTotals.each { postedTransactionTotal ->
         Map accountMap = (Map)postedTransactionTotalsMap.get(postedTransactionTotal.glAccountId)
         if (!accountMap) {
             accountMap = UtilMisc.makeMapWritable(postedTransactionTotal)
-            accountMap.put("D", BigDecimal.ZERO)
-            accountMap.put("C", BigDecimal.ZERO)
+            accountMap.put('D', BigDecimal.ZERO)
+            accountMap.put('C', BigDecimal.ZERO)
         }
         UtilMisc.addToBigDecimalInMap(accountMap, postedTransactionTotal.debitCreditFlag, postedTransactionTotal.amount)
         postedTransactionTotalsMap.put(postedTransactionTotal.glAccountId, accountMap)
-        BigDecimal debitAmount = (BigDecimal)accountMap.get("D")
-        BigDecimal creditAmount = (BigDecimal)accountMap.get("C")
+        BigDecimal debitAmount = (BigDecimal)accountMap.get('D')
+        BigDecimal creditAmount = (BigDecimal)accountMap.get('C')
         BigDecimal balance = debitAmount.subtract(creditAmount)
-        accountMap.put("balance", balance)
+        accountMap.put('balance', balance)
         glAccountCategories.each { glAccountCategory ->
-            glAccountCategoryMember = from("GlAccountCategoryMember").where("glAccountCategoryId", glAccountCategory.glAccountCategoryId, "glAccountId", postedTransactionTotal.glAccountId).orderBy("glAccountCategoryId").filterByDate().queryFirst()
+            glAccountCategoryMember = from('GlAccountCategoryMember')
+                    .where('glAccountCategoryId', glAccountCategory.glAccountCategoryId, 'glAccountId', postedTransactionTotal.glAccountId)
+                    .orderBy('glAccountCategoryId').filterByDate().queryFirst()
             if (glAccountCategoryMember) {
                 BigDecimal glAccountCategorySharePercentage = glAccountCategoryMember.amountPercentage
                 if (glAccountCategorySharePercentage && glAccountCategorySharePercentage != BigDecimal.ZERO ) {
```

Comment:
Is BigDecimal.ZERO needed here ?

Reply:
Hello @JacquesLeRoux , no, it is not needed, but it is not concerned by a codenarc rule :).

Reply:
If that is ok with you we can resolve the conversation ?